watcher/watcher.go

```diff
@@ -240,9 +256,113 @@ func (w *Watcher) reportRoomsStatusesRoutine() {
 func (w *Watcher) watchRooms() error {
 	w.RemoveDeadRooms()
 	w.AutoScale()
+	w.AddUtilizationMetricsToRedis()
 	return nil
 }
 
+// AddUtilizationMetricsToRedis store the pods usage metrics (cpu and mem) in a redis sorted set
+func (w *Watcher) AddUtilizationMetricsToRedis() {
+	logger := w.Logger.WithFields(logrus.Fields{
+		"executionID": uuid.NewV4().String(),
+		"operation":   "addUtilizationMetricsToRedis",
+		"scheduler":   w.SchedulerName,
+	})
+	logger.Info("starting to add utilization metrics to redis")
+
+	scheduler, _, _, err := controller.GetSchedulerScalingInfo(
+		logger,
+		w.MetricsReporter,
+		w.DB,
+		w.RedisClient.Client,
+		w.SchedulerName,
+	)
+
+	if err != nil {
+		if strings.Contains(err.Error(), "not found") {
+			w.Run = false
+			return
+		}
+		logger.WithError(err).Error("failed to get scheduler scaling info")
+		return
+	}
+
+	sp := scheduler.GetAutoScalingPolicy()
+	metricsMap := map[models.AutoScalingPolicyType]bool{}
+	metricsTriggers := append(sp.Up.MetricsTrigger, sp.Down.MetricsTrigger...)
+	for _, trigger := range metricsTriggers {
+		if models.ResourcePolicyType(trigger.Type) {
+			metricsMap[trigger.Type] = true
+		}
+	}
+
+	if len(metricsMap) == 0 {
+		return
+	}
+
+	requests := scheduler.GetResourcesRequests()
+	var pmetricsList *v1beta1.PodMetricsList
+	var pods *v1.PodList
+	err = w.MetricsReporter.WithSegment(models.SegmentMetrics, func() error {
+		var err error
+		pmetricsList, err = w.KubernetesMetricsClient.Metrics().PodMetricses(w.SchedulerName).List(metav1.ListOptions{})
+		return err
+	})
+	if err != nil {
+		logger.WithError(err).Error("failed to list pods metricses")
+		return
+	}
+
+	// cant get metrics from pods yet. List pods created and set usage=math.MaxInt64 for them
```

Comment:
Will this be applied to all rooms in a scheduler? I mean, if we have 9 rooms with metrics and 1 without them, will we set MaxInt64 only for the one without metrics?

Reply:
No. If we have 9 rooms with metrics and 1 without, it will only save the 9 rooms with metrics.
It will set MaxInt64 only if we have 0 rooms with metrics. It can happen if all rooms have recently started.

Maybe it is better to return an empty list when there is no room with metrics.

Do you think it is necessary to return all rooms and fill the ones without metrics with math.MaxInt64?

Reply:
It depends on how long it takes to start collecting the metrics. My concern is that we'll have ready rooms that won't be retrievable because we still don't have their metrics. If we use `math.MaxInt64` we put them with less priority but we still can use them if necessary.